### PR TITLE
Update is_fixed_role column in babelfish catalog for logins and use

### PR DESCRIPTION
**Pull request by shalinilohia50**
### Description

Update the is_fixed_role column in babelfish catalogs `babelfish_authid_login_ext` and `babelfish_authid_user_ext` for fixed database and server roles.

### Issues Resolved

Task: BABEL-5333
Signed-off-by: Shalini Lohia <lshalini@amazon.com>

### Test Scenarios Covered ###
* **Use case based -**

* **Boundary conditions -**

* **Arbitrary inputs -**

* **Negative test cases -**

* **Minor version upgrade tests -**

* **Major version upgrade tests -**

* **Performance tests -**

* **Tooling impact -**

* **Client tests -**

### Check List
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: contrib/babelfishpg_tsql/src/rolecmds.c [hunk 1]
```diff
@@ -1254,7 +1257,10 @@ add_to_bbf_authid_user_ext(const char *user_name,
 	else
 		new_record_user_ext[USER_EXT_TYPE] = CStringGetTextDatum("S");
 	new_record_user_ext[USER_EXT_OWNING_PRINCIPAL_ID] = Int32GetDatum(-1);	/* placeholder */
-	new_record_user_ext[USER_EXT_IS_FIXED_ROLE] = Int32GetDatum(-1);	/* placeholder */
+	if (strcmp(orig_user_name, DBO) != 0 && IS_FIXED_DB_PRINCIPAL(orig_user_name))
```
Comment (HarshLunagariya) [inline contrib/babelfishpg_tsql/src/rolecmds.c]: Please use strncmp.

File: test/JDBC/expected/datareader_datawriter.out [hunk 2]
```diff
@@ -239,19 +239,19 @@ go
 alter role db_datawriter add member db_roles_r1; 
 go
 
-select count(*) from sys.database_principals where name like '%db_datareader%';
+select name, is_fixed_role from sys.database_principals where name like '%db_datareader%' order by name;
 go
```
Comment (HarshLunagariya) [inline test/JDBC/expected/datareader_datawriter.out:243]: 1. Were there any other system objects which shows is_fixed_role metadata?
2. There is a possibility of refactoring of other system objects using this is_fixed_role column. We might get some perf. boost there. Let's track it separately.

File: contrib/babelfishpg_tsql/src/rolecmds.c (hunk 1 above)
Comment (rishabhtanwar29) [inline contrib/babelfishpg_tsql/src/rolecmds.c]: Question: Why `IS_FIXED_DB_PRINCIPAL` marks dbo as a fixed role if it is not actually a fixed role from TSQL perspective?

**Review by rishabhtanwar29 — DISMISSED** [review]

File: test/JDBC/expected/datareader_datawriter.out (hunk 2 above)
Comment (shalinilohia50) [reply test/JDBC/expected/datareader_datawriter.out:243]: 1. Only server_principals and database_principals shows `is_fixed_role` metadata.
2. Will create a JIRA to track it.

File: contrib/babelfishpg_tsql/src/rolecmds.c (hunk 1 above)
Comment (shalinilohia50) [reply contrib/babelfishpg_tsql/src/rolecmds.c]: Ideally `dbo` is a  fixed role and `IS_FIXED_DB_PRINCIPAL` is an internal MACRO used at multiple places where the functionality matches for `dbo` and other database roles. So, lets keep it that way.
But in sys.database_principals view, TSQL doesn't show `dbo` as a fixed role.

File: contrib/babelfishpg_tsql/src/pltsql.h [hunk 3]
```diff
@@ -2004,18 +2004,18 @@ extern bool insert_bulk_check_constraints;
...
-	 strcmp(rolname, DB_DATAREADER) == 0 || \
-	 strcmp(rolname, DB_DATAWRITER) == 0 || \
-	 strcmp(rolname, DB_DDLADMIN) == 0)
+	((strlen(rolname) == 3 && strncmp(rolname, DBO, 3) == 0) || \
+	 (strlen(rolname) == 8 && strncmp(rolname, DB_OWNER, 8) == 0) || \
+	 (strlen(rolname) == 14 && strncmp(rolname, DB_ACCESSADMIN, 14) == 0) || \
+	 (strlen(rolname) == 16 && strncmp(rolname, DB_SECURITYADMIN, 16) == 0) || \
+	 (strlen(rolname) == 13 && strncmp(rolname, DB_DATAREADER, 13) == 0) || \
+	 (strlen(rolname) == 13 && strncmp(rolname, DB_DATAWRITER, 13) == 0) || \
+	 (strlen(rolname) == 11 && strncmp(rolname, DB_DDLADMIN, 11) == 0))
```
Comment (tanscorpio7) [inline contrib/babelfishpg_tsql/src/pltsql.h:2018]: Why this change ?

**Review by tanscorpio7 — CHANGES_REQUESTED** [review]

File: contrib/babelfishpg_tsql/src/pltsql.h (hunk 3 above)
Comment (shalinilohia50) [reply contrib/babelfishpg_tsql/src/pltsql.h:2018]: It is recommended to use strncmp instead of strcmp.

**Review by rishabhtanwar29 — APPROVED** [review]

File: contrib/babelfishpg_tsql/src/pltsql.h (hunk 3 above)
Comment (tanscorpio7) [reply contrib/babelfishpg_tsql/src/pltsql.h:2018]: I cannot understand the reasoning behind this. postgres code has lots of uses to strcmp.

**Review by thephantomthief — APPROVED** [review]

File: contrib/babelfishpg_tsql/src/pltsql.h (hunk 3 above)
Comment (thephantomthief) [reply contrib/babelfishpg_tsql/src/pltsql.h:2018]: For comparing with constant strings, strcmp() is still safe
